src/cairoWriter.ts

```diff
@@ -737,10 +737,15 @@ class FunctionCallWriter extends CairoASTNodeWriter {
             nodeType instanceof UserDefinedType &&
             nodeType.definition instanceof ContractDefinition
           ) {
+            const currentContractName = node.getClosestParentByType(ContractDefinition)?.name;
             const contractType = nodeType.definition.name;
             const memberName = node.vExpression.memberName;
             const contract = writer.write(node.vExpression.vExpression);
-            return [`${contractType}.${memberName}(${contract}${args ? ', ' : ''}${args})`];
+            return [
+              `${contractType}${
+                currentContractName === contractType ? '_interface_for_this' : ''
```

Comment:
I'm not a fan of the 'interface_for_this' appearing across the codebase as string literals since it might change depending on mangling. Can it be encapsulated into a mangleOwnContractInterface function or similar, that gets shared between the files that use it?

Reply:
I'm not sure what I've done matches what'd you had expect, could you see it?